In [ ]:
"""
Custom cell to save collab data for original sims
inserted into the notebook at cell 8 on 10/02/2023 by Vishal Jain
https://colab.research.google.com/github/MilesCranmer/symbolic_deep_learning/blob/master/GN_Demo_Colab.ipynb#scrollTo=_iWH9BfUZl-r
"""
from simulate import SimulationDataset
from google.colab import drive
import numpy as np

drive.mount("/content/drive")
SEED = 42
ns = 5
for ss in sim_sets:
    sim = ss["sim"]
    if sim in ["r1", "r2", "charge", "spring", "damped", "discontinous"]:
        for n in ss["n"]:
            for dim in ss["dim"]:
                dt = ss["dt"][0]
                nt = ss["nt"][0]
                s = SimulationDataset(
                    sim, n=n, dim=dim, nt=nt, dt=dt
                )  # nt//20 datasets reduced to 1/40th of the original size for memory reasons
                title = (
                    "colab_sim={}_key={}_ns={}_n={}_dim={}_nt={}_dt={}".format(
                        sim, SEED, ns, n, dim, nt, dt
                    )
                )
                print("Running on", title)
                s.simulate(ns, key=SEED)
                np.save(
                    f"drive/MyDrive/mphil_project/data/original_colab_data/{title}.npy",
                    s.data,
                )
                print("Saved run!")

ModuleNotFoundError: No module named 'google'

In [3]:
import numpy as np

colab_data_path = "/Users/vishaljain/project/colab_sim_data/colab_sim=r2_key=42_ns=5_n=4_dim=2_nt=1000_dt=0.001.npy"
new_data_path = "/Users/vishaljain/project/sim_data/local_sim=r2_key=42_ns=5_n=4_dim=2_nt=1000_dt=0.001.npy"

colab_data = np.load(colab_data_path)
new_data = np.load(new_data_path)

# loop over time axis for a single sim and check if they are equal
for i in range(colab_data.shape[1]):
    if np.allclose(colab_data[:, i], new_data[:, i]):
        i += 1
    else:
        print("Data not equal at time step", i)
        print("Colab data for first body", colab_data[:, i, 0])
        print("New data for first body", new_data[:, i, 0])
        break

Data not equal at time step 159
Colab data for first body [[ 1.3042164   0.9814258   0.37747878 -0.41222638 -0.20222653  0.18271136]
 [-1.197527   -0.47191486  0.7170125  -1.9535376  -1.1005951   2.7955315 ]
 [ 0.3033802  -0.4703089   0.5919484   0.5930488  -1.6809258   1.0358384 ]
 [ 0.3013755   1.1393157   0.28055638  0.8894094   1.1677014   0.10828348]
 [ 2.2689323  -0.09542692 -1.4574335  -0.05348722  0.5716415   1.135643  ]]
New data for first body [[ 1.3042166   0.98142725  0.37747848 -0.41222635 -0.20222653  0.18271136]
 [-1.197527   -0.47191486  0.7170125  -1.9535376  -1.1005951   2.7955315 ]
 [ 0.3033802  -0.4703089   0.5919484   0.5930488  -1.6809258   1.0358384 ]
 [ 0.3013755   1.1393157   0.28055638  0.8894094   1.1677014   0.10828348]
 [ 2.2689323  -0.09542692 -1.4574335  -0.05348722  0.5716415   1.135643  ]]


In [15]:
import numpy as np

"""
Return the relative difference for each sim step

"""
# Define the base paths for colab and local data
colab_base_path = (
    "/Users/vishaljain/project/simulations/test_local_sim_data/local_"
)
local_base_path = (
    "/Users/vishaljain/project/simulations/test_docker_sim_data/docker_"
)

# Define the variations in simulation parameters
sims = ["r1", "r2", "charge", "damped", "spring"]
dims = [2, 3]
ns = [4, 8]

# Iterate over each combination of simulation parameters
for sim in sims:
    for dim in dims:
        for n in ns:
            # Construct the file name for each combination
            file_name = f"sim={sim}_key=42_ns=5_n={n}_dim={dim}_nt=1000_dt="
            if sim == "damped":
                dt = "2e-02" if dim == 2 else "2e-02"
            elif sim == "discontinuous" or sim == "spring":
                dt = "1e-02"
            elif sim == "r1":
                dt = "5e-03"
            else:  # For r2 and charge simulations
                dt = "1e-03"

            # Complete file paths
            colab_data_path = colab_base_path + file_name + dt + ".npy"
            new_data_path = local_base_path + file_name + dt + ".npy"

            # Load the data
            colab_data = np.load(colab_data_path)
            new_data = np.load(new_data_path)

            # Loop over the time axis for a single sim and check if they are equal
            for i in range(colab_data.shape[1]):
                if not np.allclose(colab_data[0, i], new_data[0, i]):
                    print(
                        f"Data not equal at time step {i} for sim={sim}, n={n}, dim={dim}"
                    )
                    print(
                        "Colab data for first body", colab_data[0, i, 0]
                    )  # Assuming you want to print the first body
                    print("New data for first body", new_data[0, i, 0])
                    print("\n")
                    break

Data not equal at time step 661 for sim=r1, n=8, dim=2
Colab data for first body [-2.2167478  -2.1893718  -0.85543907  0.7472024   0.04691693  0.4549926 ]
New data for first body [-2.2167492  -2.1893697  -0.8554389   0.7472033   0.04691693  0.4549926 ]


Data not equal at time step 58 for sim=charge, n=4, dim=2
Colab data for first body [ 1.2631352   1.0166491   0.2500642  -0.15758552 -1.          0.18271136]
New data for first body [ 1.2631342   1.0166492   0.25006413 -0.15758543 -1.          0.18271136]




Data not equal at time step 661 for sim=r1, n=8, dim=2
Colab data for first body [-2.2167492  -2.1893697  -0.8554389   0.7472033   0.04691693  0.4549926 ]
New data for first body [-2.2167478  -2.1893718  -0.85543907  0.7472024   0.04691693  0.4549926 ]


Data not equal at time step 58 for sim=charge, n=4, dim=2
Colab data for first body [ 1.2631342   1.0166492   0.25006413 -0.15758543 -1.          0.18271136]
New data for first body [ 1.2631352   1.0166491   0.2500642  -0.15758552 -1.          0.18271136]



In [18]:
data_1 = np.load(
    "/Users/vishaljain/project/simulations/test_local_sim_data/local_sim=r2_key=42_ns=5_n=4_dim=2_nt=1000_dt=1e-03.npy"
)
data_2 = np.load(
    "/Users/vishaljain/project/test_cli/sim=r2_ns5_seed42_n_body=4_dim=2_nt=1000_dt=1e-03_data.npy"
)
np.allclose(data_1, data_2)

TypeError: expected str, bytes or os.PathLike object, not set